Lib/site.py

```diff
@@ -390,6 +390,23 @@ def setcopyright():
         files, dirs)
 
 
+def _register_detect_pip_usage_in_repl():
+    old_excepthook = sys.excepthook
+
+    def detect_pip_usage_in_repl(typ, value, traceback):
+        if typ is SyntaxError and (
+            "pip install" in value.text or "pip3 install" in value.text
+        ):
+            value.msg = ("The Python package manager (pip) can only be used"
```

Comment:
While we could potentially make this a module level constant (so it's easier for alternate shells to customise the message without needing to replace the entire hook), I think it will instead be better to expose the original wrapped hook on the wrapper implementation.

Reply:
Assuming your conclusion here is basically the same as your next comment, this should be addressed in 2d302e6cd1ff0fc058d14b7583c8352d5125640d.